pkg/apis/cloudcore/v1alpha1/validation/validation.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2019 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+	"path"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	cloudconfig "github.com/kubeedge/kubeedge/pkg/apis/cloudcore/v1alpha1"
+	utilvalidation "github.com/kubeedge/kubeedge/pkg/util/validation"
+)
+
+// ValidateCloudCoreConfiguration validates `c` and returns an errorList if it is invalid
+func ValidateCloudCoreConfiguration(c *cloudconfig.CloudCoreConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, ValidateKubeAPIConfig(*c.KubeAPIConfig)...)
+	allErrs = append(allErrs, ValidateModuleCloudHub(*c.Modules.CloudHub)...)
+	allErrs = append(allErrs, ValidateModuleEdgeController(*c.Modules.EdgeController)...)
+	allErrs = append(allErrs, ValidateModuleDeviceController(*c.Modules.DeviceController)...)
+	return allErrs
+}
+
+// ValidateModuleCloudHub validates `c` and returns an errorList if it is invalid
+func ValidateModuleCloudHub(c cloudconfig.CloudHub) field.ErrorList {
+	if !c.Enable {
+		return field.ErrorList{}
+	}
+
+	allErrs := field.ErrorList{}
+	validWPort := utilvalidation.IsValidPortNum(int(c.WebSocket.Port))
+	validAddress := utilvalidation.IsValidIP(c.WebSocket.Address)
```

Comment:
should validate if Address is valid host here: user may like to use hostname or domain name

Reply:
No worries, at serverside, in most cases, process should listen on an ip address